### PR TITLE
Add open user data button to main menu

### PR DESCRIPTION
**Pull request by rubenwardy**
As suggested by sfan5. This is in the credits menu because there it shouldn't be heavily promoted to users, and there's also very little space on the content tab. This can be moved later, I don't particularly feel like redesigning the content tab

image

## To do

- [ ] Test on Windows and macOS

### CODE REVIEW

File: src/porting.cpp [hunk 1]
```diff
@@ -719,25 +719,30 @@ int mt_snprintf(char *buf, const size_t buf_size, const char *fmt, ...)
...
 }
 
-bool openURL(const std::string &url)
+inline bool checkIsURL(const std::string &uri)
 {
-	if ((url.substr(0, 7) != "http://" && url.substr(0, 8) != "https://") ||
-			url.find_first_of("\r\n") != std::string::npos) {
-		errorstream << "Invalid url: " << url << std::endl;
+	return (uri.substr(0, 7) == "http://" || uri.substr(0, 8) == "https://") &&
+			uri.find_first_of("\r\n") == std::string::npos;
```
Comment (sfan5) [inline src/porting.cpp:725]: the \r\n check must be done regardless of URL or not

File: src/porting.cpp [hunk 2]
```diff
@@ -719,29 +719,48 @@ int mt_snprintf(char *buf, const size_t buf_size, const char *fmt, ...)
...
+		return false;
+	}
+
+	return open_uri(url);
+}
+
+bool open_directory(const std::string &path)
+{
+	if (!fs::IsDir(path)) {
+		errorstream << "Unable to open directory as it does not exist. Path=" << path << std::endl;
```
Comment (sfan5) [inline src/porting.cpp:757]:
```suggestion
		errorstream << "Unable to open directory as it does not exist: " << path << std::endl;
```

**Review by LoneWolfHT — APPROVED** [review]
Works on Windows with my installer. This will be so useful

**Review by SmallJoker — APPROVED** [review]
Works on Linux/RUN_IN_PLACE=1. I assume you already tested that, but considering there's also a successful test for Windows it works well enough.

**Pull request merged**